e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/gef/deal-cancellation-maker-unable-issue-facility.spec.js

```diff
@@ -0,0 +1,151 @@
+import { DEAL_STATUS, DEAL_SUBMISSION_TYPE } from '@ukef/dtfs2-common';
+import gefPages from '../../../../../../../gef/cypress/e2e/pages';
+import MOCK_USERS from '../../../../../../../e2e-fixtures/portal-users.fixture';
+import { MOCK_APPLICATION_AIN_DRAFT, MOCK_APPLICATION_MIN_DRAFT } from '../../../../../../../e2e-fixtures/gef/mocks/mock-deals';
+import { anIssuedCashFacility, anUnissuedCashFacility, multipleMockGefFacilities } from '../../../../../../../e2e-fixtures/mock-gef-facilities';
+import { yesterday, tomorrow } from '../../../../../../../e2e-fixtures/dateConstants';
+
+import { TFM_URL, PIM_USER_1 } from '../../../../../../../e2e-fixtures';
+
+const { BANK1_MAKER1 } = MOCK_USERS;
+
+context('When a deal has been cancelled on TFM, maker unable to issue facility', () => {
+  const ainDeals = Array(4).fill(MOCK_APPLICATION_AIN_DRAFT);
+  const minDeals = Array(4).fill(MOCK_APPLICATION_MIN_DRAFT);
+  const gefDeals = [...ainDeals, ...minDeals];
+  const deals = [];
+
+  const facilities = [
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    anUnissuedCashFacility({ facilityEndDateEnabled: true }),
+    multipleMockGefFacilities({ facilityEndDateEnabled: true }).unissuedContingentFacility,
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    multipleMockGefFacilities({ facilityEndDateEnabled: true }).unissuedContingentFacility,
+    anUnissuedCashFacility({ facilityEndDateEnabled: true }),
+  ];
+
+  before(() => {
+    cy.insertManyGefDeals(gefDeals, BANK1_MAKER1).then((insertedDeals) => {
+      insertedDeals.forEach((insertedDeal, index) => {
+        const gefDeal = { ...insertedDeal };
+
+        // updates a gef deal so has relevant fields
+        cy.updateGefDeal(gefDeal._id, index < 4 ? MOCK_APPLICATION_AIN_DRAFT : MOCK_APPLICATION_MIN_DRAFT, BANK1_MAKER1);
```

Comment:
I think its still a bit unintuitive to have these ternaries in the test set up but happy to leave for now

Reply:
Could we have this as a `mockDeal` const please? I think would be clearer

```
const mockDeal = index < 4 ? MOCK_APPLICATION_AIN_DRAFT : MOCK_APPLICATION_MIN_DRAFT;
```

Is there are reason for index less than 4? Please ignore if this is an existing thing.